### DISCUSSION TITLE
Add component categorization of an existing component

### DISCUSSION THREAD

**Discussion by tomkolp** [WeblateOrg/weblate#9807] [General] [answered]
Can I add existing components to the "component categorization" tree? I have over 2000 components in the project so far I have categorized them by component lists. Component lists must remain because all permission rules are based on them. Can I add existing components to the category tree? I don't see that option anywhere.

**Comment by nijel** [comment]
Yes, in manage / move.  Or using API.

**Comment by tomkolp** [comment]
I can't find such an option
![image](https://github.com/WeblateOrg/weblate/assets/39633242/8f8eb543-c352-4ae1-9c31-39bdd49c8f97)
![image](https://github.com/WeblateOrg/weblate/assets/39633242/e1d97845-2fe3-4913-9a8a-759b1dda3e1a)

<html>
<body>


5.0.1-dev — 294c73e6f8d37b1194d2ecd7f44a00f30652c3a3
--



</body>
</html>5.0.1-dev — [294c73e6f8d37b1194d2ecd7f44a00f30652c3a3](https://github.com/WeblateOrg/weblate/commits/294c73e6f8d37b1194d2ecd7f44a00f30652c3a3)

**Reply by nijel** [reply]
On the component you want to move

**Comment by tomkolp** [comment]
I searched everywhere
![Screenshot_20230825_202432_Edge](https://github.com/WeblateOrg/weblate/assets/39633242/e31736dc-4b17-48e9-88e8-b801b726ec77)

**Reply by nijel** [reply, marked as answer]
"Rename" is the item you want to use here. It should be instead called "Rename or move" or something like that...

**Reply by nijel** [reply]
See https://github.com/WeblateOrg/weblate/pull/9880

**Reply by tomkolp** [reply]
Cannot rename due to outstanding issue in the configuration: {'check_flags': ['Invalid translation flag: "ignore-xml-invalid"']}

**Reply by nijel** [reply]
Did you disable that check? https://docs.weblate.org/en/latest/user/checks.html#xml-syntax

**Reply by tomkolp** [reply]
yes throughout the project.
![image](https://github.com/WeblateOrg/weblate/assets/39633242/014c4d2b-05fa-4b5f-bdff-375bdd33a12a)

**Reply by nijel** [reply]
This more looks like it is disabled site-wide using WEBLATE_REMOVE_CHECK, the component configuration needs to be adjusted then.

**Reply by tomkolp** [reply]
After customizing the component, I can change the category.
I noticed that after adding a component to a category, all links of the component change.
I have scripts to download 400 componets via api, I will have to adapt them.
I think you need to add a warning about changing the structure of links, e.g. in the documentation.

**Reply by nijel** [reply]
The warning is already present in the form:

![image](https://github.com/WeblateOrg/weblate/assets/212189/7534226e-573a-4def-86ba-0adf9ba7934c)

I've just added more prominent display in 44d8aae28143e6825258f1cbfabd0d4e95d8a1a1.

> I have scripts to download 400 componets via api, I will have to adapt them.

Why not using https://docs.weblate.org/en/latest/api.html#get--api-projects-(string-project)-components- to list all components in the project?

**Reply by tomkolp** [reply]
Because I only need components from a specific "component list" and not the whole project.

**Reply by nijel** [reply]
https://github.com/WeblateOrg/weblate/pull/9882 further clarifies the warning.

**Reply by nijel** [reply]
> Because I only need components from a specific "component list" and not the whole project.

Ah, GET is missing there, added in https://github.com/WeblateOrg/weblate/pull/9883

**Reply by tomkolp** [reply]
Nice :)

**Reply by tomkolp** [reply]
The last question. Assigning several dozen components to a category triggered some database task that blocks the upload of changes to the github repo. Can I keep docker containers? Will the task resume after reboot?

Today I scheduled physical server maintenance and I can't reschedule it.

**Reply by nijel** [reply]
Can you check what kind of tasks is that? It could be just a metrics update and these would be calculated again if needed.

**Reply by tomkolp** [reply]
@nijel 
I have about 3000 components in the project, I added about 300 to categories.

When I open the project, the added components are correctly in the categories, but also in the "project main folder".

Is this the intended action?

Next, the project is loaded, which has 30 pages. To find out whether a given component already has a category, you need to check the link of this component.

I thought that the entire category structure would work like a folder structure, and moving a component would be cut->paste, not the current "copy->paste" way.

**Reply by nijel** [reply]
That's bug, should be fixed by https://github.com/WeblateOrg/weblate/pull/9912
